### PR TITLE
(feat): Distribute describe_objects S3 method

### PR DESCRIPTION
**Pull request by jaidisido**
### Feature or Bugfix

- Feature

### Detail
- Distribute describe_objects S3 method with executor

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: awswrangler/s3/_read.py [hunk 1]
```diff
@@ -110,28 +107,3 @@ def _union(dfs: List[pd.DataFrame], ignore_index: Optional[bool]) -> pd.DataFram
         for df in dfs:
             df[col] = pd.Categorical(df[col].values, categories=cat.categories)
     return pd.concat(objs=dfs, sort=False, copy=False, ignore_index=ignore_index)
-
-
-def _read_dfs_from_multiple_paths(
```
Comment (jaidisido) [inline awswrangler/s3/_read.py:115]: This method is not use anywhere anymore

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-s6u9F3qN9oFy
* Commit ID: 8f9444a61ecd7e63ffe6fc23b615f8afd218060c
* Result: SUCCEEDED
* [Build Logs](https://pciyom2pbh.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=da6dd4cc-981e-40c3-b7b9-e8e82bb6ce6d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by kukushking — APPROVED** [review]
Nice one!

**Review by LeonLuttenberger — APPROVED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubStandardCodeBuild8C06-llutOAimTATs
* Commit ID: 8f9444a61ecd7e63ffe6fc23b615f8afd218060c
* Result: FAILED
* [Build Logs](https://jsv9u8gote.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1b215589-1093-4b71-8dd1-8965cf74a7e8%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by malachi-constant — APPROVED** [review]

**Pull request merged**